snapcraft_legacy/storeapi/v2/validation_sets.py

```diff
@@ -14,19 +14,21 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from typing import Any, Annotated, Dict, Literal, TYPE_CHECKING
+from typing import Any, Literal, TYPE_CHECKING
 import numbers
 import pydantic
 
 from craft_application import models
+from pydantic import StringConstraints
+from typing_extensions import Annotated
 
 
 if TYPE_CHECKING:
     SnapName = str
     SnapId = str
 else:
-    SnapName = pydantic.constr(max_length=40)
-    SnapId = pydantic.constr(max_length=40)
+    SnapName = Annotated[str, StringConstraints(max_length=40)]
```

Comment:
The `TYPE_CHECKING` conditional is no longer necessary when you use `Annotated`.